### DISCUSSION TITLE
Windows CI, slow tests

### DISCUSSION THREAD

**Discussion by icing** [curl/curl#14849] [Q&A]
I looked a bit into the often failing Windows CI jobs. It's a bit of a mystery. We flagged the test cases that are unreliable on this platform and the situation now is that jobs timeout after 14 minutes. The reason for this is that there are several test cases that do not fail, but are very slow, leading to an overall delay that triggers the abort.

Sample of slow test cases below (these are only the worst). These run at a fraction on other platforms.

Since I cannot run these locally on a Windows machine, can someone provide detailed logs of these that may shed some light on where the time is spent?

```
2038: 18.398s
2037: 18.561s
3001: 18.615s
2034: 18.67s
2041: 18.694s
2042: 18.712s
1939: 19.942s
1273: 21.104s
1501: 21.95s
1561: 22.494s
313: 22.697s
1238: 23.398s
445: 24.285s
1117: 25.048s
1523: 25.798s
303: 28.382s
3207: 28.663s
1086: 34.121s
1112: 34.74s
```

**Comment by icing** [comment]
What might be interesting to investigate here:
* the result are from a run with parallel execution, e.g. `curl > cd tests; ./runtests.pl -j20`. Is the job scheduling just bad or hands under certain conditions?
* or are individual tests really that slow? What does a single test give? Like: `curl > cd tests; ./runtests.pl -v 1112`

**Comment by vszakats** [comment]
Though not all of these are TLS-related, I think we did not raise the timeouts enough when adding TLS tests recently in bc72a78a11764558639131d51fa8e7c81ee1e113 #14488. It's not solving the root cause of course, but bumping the timeouts more would be nice → #14851.

**Reply by vszakats** [reply]
Continuing with the hang/other issues here: #14854
